src/main/java/bio/terra/service/snapshotbuilder/CriteriaQueryUtils.java

```diff
@@ -0,0 +1,146 @@
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FieldPointer;
+import bio.terra.service.snapshotbuilder.query.FieldVariable;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Literal;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.BinaryFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.BooleanAndOrFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.FunctionFilterVariable;
+import bio.terra.service.snapshotbuilder.query.filtervariable.SubQueryFilterVariable;
+import java.util.List;
+import org.apache.arrow.util.VisibleForTesting;
+
+public class CriteriaQueryUtils {
+  @VisibleForTesting
+  public static BooleanAndOrFilterVariable generateFilterForRangeCriteria(
+      SnapshotBuilderProgramDataRangeCriteria rangeCriteria) {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+    return new BooleanAndOrFilterVariable(
+        BooleanAndOrFilterVariable.LogicalOperator.AND,
+        List.of(
+            new BinaryFilterVariable(
+                new FieldVariable(
+                    new FieldPointer(tablePointer, rangeCriteria.getName()), tableVariable),
+                BinaryFilterVariable.BinaryOperator.GREATER_THAN_OR_EQUAL,
+                new Literal(rangeCriteria.getLow().doubleValue())),
+            new BinaryFilterVariable(
+                new FieldVariable(
+                    new FieldPointer(tablePointer, rangeCriteria.getName()), tableVariable),
+                BinaryFilterVariable.BinaryOperator.LESS_THAN_OR_EQUAL,
+                new Literal(rangeCriteria.getHigh().doubleValue()))));
+  }
+
+  @VisibleForTesting
+  public static FunctionFilterVariable generateFilterForListCriteria(
+      SnapshotBuilderProgramDataListCriteria listCriteria) {
+    TablePointer tablePointer = TablePointer.fromTableName("person");
+    TableVariable tableVariable = TableVariable.forPrimary(tablePointer);
+    TablePointer conceptPointer = TablePointer.fromTableName("concept");
+    TableVariable conceptPointerVariable =
+        TableVariable.forJoined(
+            conceptPointer,
+            "concept_id",
+            new FieldVariable(
+                new FieldPointer(tablePointer, listCriteria.getName()), tableVariable));
+    return new FunctionFilterVariable(
+        FunctionFilterVariable.FunctionTemplate.IN,
+        new FieldVariable(new FieldPointer(conceptPointer, "concept_id"), conceptPointerVariable),
+        listCriteria.getValues().stream()
+            .map(value -> new Literal(value.doubleValue()))
```

Comment:
I think we made a mistake here, that snapshot builder IDs in general should be `integer` not `number` in the openapi, so that IDs like this should be `Integer` not `BigDecimal`. Maybe use `value.intValue()` for now, or PR another change to fix the ID type in the yaml